src/pytest_mock_resources/fixture/database/relational/async_/generic.py

```diff
@@ -0,0 +1,188 @@
+import abc
```

Comment:
After reading through this, it's not entirely obvious to me whether/why we should need to actually use async from within the actual engine setup code.

I'll prefix my suggestion with the only real caveat that i can think of, which is: 
you'll need both psycopg2 as well as asyncpg as dev requirements, whereas with a fully native async implementation you can get away with just asyncpg (but the requirement on pytest-asyncio). Although i guess you didn't change the docker connection checking logic, so this will still require psycopg2 today anyways.

Okay, with that aside, i feel like you could drastically simplify your PR by just doing all the pre-test stuff with the sync version, and just yielding the async engine (without actually having connected to it).

That is to say something like:
```
engine_manager = EngineManager(engine, ordered_actions, tables=tables, default_schema="public")
yield from engine_manager.manage_async(session=session)  # this was previously that wonky `for` loop for python2 backwards compat, nowaways you can just `yield from`
```

but internally, you could still use `engine` to i.e. `self._run_actions()`. Then afterward, `EngineManager.manage_async` could do the session/engine management code you're already doing.